durablefunctionsmonitor.dotnetisolated.core/Common/Auth.cs

```diff
@@ -117,22 +117,12 @@ public static async Task<DfmMode> ValidateIdentityAsync(HttpRequestData request,
             // Then validating anti-forgery token
             ThrowIfXsrfTokenIsInvalid(request.Headers, request.Cookies);
 
-            // Need to convert Identities (of which is supposed to be just one) to ClaimsPrincipal, because the token validation routine returns ClaimsPrincipal type.
-            var principal = new ClaimsPrincipal(request.Identities);
-
-            // Trying with EasyAuth
+            var principal = await GetClaimsPrincipal(request, settings.UserNameClaimName, settings.RolesClaimName);
             var userNameClaim = principal.FindAll(settings.UserNameClaimName).SingleOrDefault();
-            if (userNameClaim == null)
-            {
-                // Validating and parsing the token ourselves
-                request.Headers.TryGetValues("Authorization", out var authHeaderValues);
-                principal = await ValidateToken(authHeaderValues?.SingleOrDefault());
-                userNameClaim = principal.FindFirst(settings.UserNameClaimName);
-            }
 
             if (userNameClaim == null)
             {
-                throw new DfmUnauthorizedException($"'{settings.UserNameClaimName}' claim is missing in the incoming identity. Call is rejected.");
+                throw new DfmUnauthorizedException($"'{settings.UserNameClaimName}' claim is missing or duplicated in the incoming identity. Call is rejected.");
```

Comment:
not fully certain I understand where the possible duplication is coming from. Could you break that down for me?

Reply:
For a brief moment something made me think that .SingleOrDefault() would return null on duplicates. Which is, of course, absolutely not the case. Will revert this.